### PR TITLE
[MODORGSTOR-143]-java 17 upgrade.

### PR DESCRIPTION
**Pull request by SinghAdes**
https://issues.folio.org/browse/MODORGSTOR-143

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate action.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code are 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added, or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildMvn {
   publishModDescriptor = 'yes'
   mvnDeploy = 'yes'
-  buildNode = 'jenkins-agent-java11'
+  buildNode = 'jenkins-agent-java17'
```
Comment (tatsiana-tarhonskaya) [inline Jenkinsfile:4]: here health check command should be updated to use wget instead of curl https://wiki.folio.org/display/TC/JDK+17+and+Java+17

Comment (SinghAdes) [reply Jenkinsfile:4]: got it, thanks

**Review by grabsefx — APPROVED** [review]

**Review by tatsiana-tarhonskaya — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
   doDocker = {
     buildJavaDocker {
       publishMaster = 'yes'
       healthChk = 'yes'
-      healthChkCmd = 'curl -sS --fail -o /dev/null  http://localhost:8081/apidocs/ || exit 1'
+      healthChkCmd = 'wget --no-verbose --tries=1 --spider http://localhost:8081/admin/health || exit 1'
```
Comment (SerhiiNosko) [inline Jenkinsfile:10]: I am not sure does vertx app has this endpoint /admin/health ?

Comment (SerhiiNosko) [reply Jenkinsfile:10]: I see it also in another vertx modules such as mod-login with java 17, so I think we are okay

**Pull request merged**